File: frontend/src/api/modules.ts
import { api } from "@/lib/api"

export type ProductModule = {
  id: string
  name: string
  icon: string
  prompt_namespace: string
  frontend_entry: string
  components: string[]
  report_modes: string[]
  has_sub_questions: boolean
  has_expert_defaults: boolean
  has_prompt_defaults: boolean
  supports_interview: boolean
}

export function listModules(): Promise<ProductModule[]> {
  return api.get<ProductModule[]>("/modules")
}

File: backend/app/schemas/modules.py
"""Serializable product-module metadata (no routers or callables)."""

from __future__ import annotations

from pydantic import BaseModel, Field


class ModuleOut(BaseModel):
    id: str
    name: str
    icon: str
    prompt_namespace: str
    frontend_entry: str
    components: list[str] = Field(default_factory=list)
    report_modes: list[str] = Field(default_factory=list)
    has_sub_questions: bool
    has_expert_defaults: bool
    has_prompt_defaults: bool = False
    supports_interview: bool
